### PR TITLE
[Breaking] Fix custom metric for multi output.

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/dmlc/xgboost/issues/5952 .

### CODE REVIEW

**Comment by trivialfis** [comment]
Removed blocking.  This will be a breaking change.

**Comment by hcho3** [comment]
@trivialfis So this is to be consistent with #5564? Can we include it in this release?

File: src/learner.cc [hunk 1]
```diff
@@ -156,7 +156,7 @@ struct LearnerTrainParam : public XGBoostParameter<LearnerTrainParam> {
   // data split mode, can be row, col, or none.
   DataSplitMode dsplit;
   // flag to disable default metric
-  int disable_default_eval_metric;
+  bool disable_default_eval_metric;
```
Comment (hcho3) [inline src/learner.cc:159]: The parameter doc should also be updated.

**Comment by trivialfis** [comment]
>  So this is to be consistent with #5564?

Yes.

> Can we include it in this release? 

Yes, I want to include it.  Just not sure about how others feel with a last minute breaking change.

> The release note can indicate that it's breaking.

Will mark it.

**Comment by hcho3** [comment]
> Just not sure about how others feel with a last minute breaking change.

In this case, we already have a precedent (#5564) where custom objectives gets the margin scores. It will be worthwhile to achieve the same behavior in custom metrics too. Also, this PR now adds unit tests for custom metrics with multiple outputs.

**Comment by trivialfis** [comment]
@hcho3 Thanks, remarked as blocking.

**Pull request merged**